lib/archethic/oracle_chain/services/uco_price.ex

```diff
@@ -13,20 +13,69 @@ defmodule Archethic.OracleChain.Services.UCOPrice do
 
   @impl Impl
   @spec fetch() :: {:ok, %{required(String.t()) => any()}} | {:error, any()}
-  def fetch, do: provider().fetch(@pairs)
+  def fetch do
+    ## retrieve prices from configured providers and filter results marked as errors
+    prices =
+      providers()
+      |> Task.async_stream(
+        fn provider ->
+          case provider.fetch(@pairs) do
+            {:ok, _prices} = result ->
+              result
+
+            {:error, reason} ->
+              Logger.warning("Service : #{inspect(__MODULE__)} : Cannot fetch values from
+                provider: #{inspect(provider)} with reason : #{inspect(reason)}.")
+
+              {false, provider}
+          end
+        end,
+        on_timeout: :kill_task
+      )
+      |> Stream.filter(fn
+        {:ok, {:ok, _result}} ->
+          true
+
+        other ->
+          Logger.warning(
+            "Service : #{inspect(__MODULE__)} : Unexpected answer while querying provider : #{inspect(other)}"
+          )
+
+          false
+      end)
+      |> Enum.map(fn
+        {_, {_, result = %{}}} ->
+          result
+
+        other_service_answer_format ->
```

Comment:
I agree, but if it comes a service returns badly formatted values, then this will crash the scheduler, I wouldn't take the risk to rely on having always valid format from queried services.

Reply:
I think the provider cannot return badly formatted value as it is defined in the `fetch` function spec of the provider, so if a provider can return a non formatted value, dialyzer will raise a warning